src/fen_util.rs

```diff
@@ -95,7 +124,28 @@ pub fn valid_fen(fen: &str) -> bool {
     true
 }
 
-/// Generates a FEN (Forsyth-Edwards Notation) string representing the current state of the chessboard.
+/// Generates a Forsyth-Edwards Notation (FEN) string representing the current state of the chessboard.
+///
+/// # Arguments
+///
+/// - `chessboard`: A reference to the `Chessboard` struct containing the current game state.
+///
+/// # Returns
+///
+/// A string representing the FEN string for the current chessboard position.
+///
+/// # Example
+///
+/// ```
+/// use kn_o3::Chessboard;
+///
+/// let mut chessboard = Chessboard::new();
+/// let fen_string = get_fen_placement(&chessboard);
+/// assert_eq!(fen_string, "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR");
+/// ```
+///
+/// The function generates a FEN string representing the current state of the chessboard.
+/// It iterates through each rank and file, encoding the piece positions and empty squares.
```

Comment:
I'd say this is pretty self explanatory too. Maybe just include the `rnbq...` in the returns section